scipy/interpolate/_bspl.pyx

```diff
@@ -300,57 +300,6 @@ def _colloc(const double[::1] x, const double[::1] t, int k, double[::1, :] ab,
                 ab[kl + ku + j + offset - clmn, clmn] = wrk[a]
 
 
-@cython.wraparound(False)
-@cython.boundscheck(False)
-def _handle_lhs_derivatives(const double[::1]t, int k, double xval,
-                            double[::1, :] ab,
-                            int kl, int ku,
-                            const cnp.npy_long[::1] deriv_ords,
-                            int offset=0):
-    """ Fill in the entries of the collocation matrix corresponding to known
-    derivatives at xval.
-
-    The collocation matrix is in the banded storage, as prepared by _colloc.
-    No error checking.
-
-    Parameters
-    ----------
-    t : ndarray, shape (nt + k + 1,)
-        knots
-    k : integer
-        B-spline order
-    xval : float
-        The value at which to evaluate the derivatives at.
-    ab : ndarray, shape(2*kl + ku + 1, nt), Fortran order
-        B-spline collocation matrix.
-        This argument is modified *in-place*.
-    kl : integer
-        Number of lower diagonals of ab.
-    ku : integer
-        Number of upper diagonals of ab.
-    deriv_ords : 1D ndarray
-        Orders of derivatives known at xval
-    offset : integer, optional
-        Skip this many rows of the matrix ab.
-
-    """
-    cdef:
-        int left, nu, a, clmn, row
-        double[::1] wrk = np.empty(2*k+2, dtype=np.float64)
-
-    # derivatives @ xval
-    with nogil:
```

Comment:
just for my understanding, could you explain the relationship between this `with nogil` code and the Python translation?

Reply:
The code is precisely the same, modulo that in python it operates on numpy arrays and in cython it was taking typed memoryviews of arrays.

And while it was in a `with nogil` section, the is doing so little work that releasing the GIL is _very_ unlikely to have any effect.